pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_service.go

```diff
@@ -152,13 +156,22 @@ func (s *service) Deploy(ctx context.Context) error {
 		if s.values.enableSNI {
 			metav1.SetMetaDataAnnotation(&obj.ObjectMeta, "networking.istio.io/exportTo", "*")
 		}
+		if s.values.topologyAwareRoutingEnabled {
+			metav1.SetMetaDataAnnotation(&obj.ObjectMeta, corev1.AnnotationTopologyAwareHints, "auto")
+		} else {
+			delete(obj.Annotations, corev1.AnnotationTopologyAwareHints)
+		}
 
 		utilruntime.Must(gardenerutils.InjectNetworkPolicyAnnotationsForScrapeTargets(obj, networkingv1.NetworkPolicyPort{Port: utils.IntStrPtrFromInt(kubeapiserver.Port), Protocol: utils.ProtocolPtr(corev1.ProtocolTCP)}))
 
 		obj.Labels = getLabels()
 		if s.values.gardenerManaged {
 			metav1.SetMetaDataLabel(&obj.ObjectMeta, v1beta1constants.LabelAPIServerExposure, v1beta1constants.LabelAPIServerExposureGardenerManaged)
 		}
+		if s.values.topologyAwareRoutingEnabled {
+			metav1.SetMetaDataLabel(&obj.ObjectMeta, resourcesv1alpha1.EndpointSliceHintsConsider, "true")
+			// No need to delete the consider label when topologyAwareRoutingEnabled=false as the labels are overwritten few lines above.
```

Comment:
Can you drop this comment and rather fix L167 to not overwrite the entire set of labels?

Reply:
I recall you shared that we should avoid overwriting all the labels/annotations to allow extensions or other components to add labels/annotations.
This topic is not that much related to this PR. It deserves a dedicated issue and a convention in `component-checklist.md`.